arch/arm/core/irq_manage.c

```diff
@@ -262,4 +262,33 @@ int arch_irq_connect_dynamic(unsigned int irq, unsigned int priority,
 	z_arm_irq_priority_set(irq, priority, flags);
 	return irq;
 }
+
+#ifdef CONFIG_DYNAMIC_DIRECT_INTERRUPTS
+static inline void z_arm_irq_dynamic_direct_isr_dispatch(void)
+{
+	u32_t irq = __get_IPSR() - 16;
+
+	if (irq < IRQ_TABLE_SIZE) {
+		struct _isr_table_entry *isr_entry = &_sw_isr_table[irq];
+
+		isr_entry->isr(isr_entry->arg);
```

Comment:
Why not do:
`return isr_entry->isr(isr_entry->arg);`
and then you don't need the `_reschedule` and `_no_reschedule` variants at all

Reply:
```
struct _isr_table_entry {
	void *arg;
	void (*isr)(void *);
};
```

Looks to me function returns void